shanoir-ng-datasets/src/main/java/org/shanoir/ng/processing/carmin/schedule/ExecutionStatusMonitor.java

```diff
@@ -0,0 +1,265 @@
+import java.nio.file.Path;
+import java.nio.file.PathMatcher;
+import java.nio.file.Paths;
+import java.util.Comparator;
+import java.util.stream.Stream;
+
+import javax.ws.rs.NotFoundException;
+
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
+import org.apache.commons.compress.archivers.tar.TarArchiveInputStream;
+import org.apache.commons.compress.compressors.gzip.GzipCompressorInputStream;
+import org.apache.commons.compress.utils.IOUtils;
+import org.shanoir.ng.dataset.modality.ProcessedDatasetType;
+import org.shanoir.ng.importer.dto.ProcessedDatasetImportJob;
+import org.shanoir.ng.importer.service.ImporterService;
+import org.shanoir.ng.processing.carmin.model.CarminDatasetProcessing;
+import org.shanoir.ng.processing.carmin.model.Execution;
+import org.shanoir.ng.processing.carmin.model.ExecutionStatus;
+import org.shanoir.ng.processing.carmin.service.CarminDatasetProcessingService;
+import org.shanoir.ng.processing.model.DatasetProcessing;
+import org.shanoir.ng.processing.service.DatasetProcessingService;
+import org.shanoir.ng.shared.exception.EntityNotFoundException;
+import org.shanoir.ng.shared.model.Study;
+import org.shanoir.ng.shared.repository.StudyRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * CRON job to request VIP api and create porcessedDataset
+ * 
+ * @author KhalilKes
+ */
+@Service
+public class ExecutionStatusMonitor implements ExecutionStatusMonitorService {
+
+    @Value("${vip.uri}")
+    private String VIP_URI;
+
+    @Value("${vip.upload-folder}")
+    private String importDir;
+
+    @Value("${vip.sleep-time}")
+    private long sleepTime;
+
+    @Value("${vip.file-formats}")
+    private String[] listOfNiftiExt;
+
+    private boolean stop;
+
+    private String identifier;
+
+    private static final Logger LOG = LoggerFactory.getLogger(ExecutionStatusMonitor.class);
+
+    @Autowired
+    private CarminDatasetProcessingService carminDatasetProcessingService;
+
+    @Autowired
+    private DatasetProcessingService datasetProcessingService;
+
+    @Autowired
+    private ImporterService importerService;
+
+    @Autowired
+    private StudyRepository studyRepository;
+
+    @Async
+    @Override
+    @Transactional
+    public void startJob(String identifier) {
+
+        this.identifier = identifier;
+        this.stop = false;
+
+        String uri = VIP_URI + identifier + "/summary";
+        RestTemplate restTemplate = new RestTemplate();
+
+        while (!stop) {
```

Comment:
Hi Khalil, this while loop is maybe not the best solution. One separate job, that is invoked in regular intervals, for each process to monitor should be used, so a method that is regularly called will check for the finished state and kill itself after instead of having a while loop that runs all the time. For each started processing. With kind regards, Michael

Reply:
the above is quite confuse, please read below